geowebcache/core/src/main/java/org/geowebcache/config/XMLConfiguration.java

```diff
@@ -1086,10 +1086,186 @@ public Boolean getfullWMS(){
     }
 
     @Override
-    public List<BlobStoreConfig> getBlobStores() {
-        return getGwcConfig().getBlobStores();
+    public List<BlobStoreInfo> getBlobStores() {
+        // need to return an unmodifiable list of unmodifiable BlobStoreInfos
+        return Collections.unmodifiableList(getGwcConfig().getBlobStores().
+            stream().map((info) -> {
+                return (BlobStoreInfo)info.clone();
+            }).collect(Collectors.toList())
+        );
     }
-    
+
+    @Override
+    public synchronized void addBlobStore(BlobStoreInfo info) {
```

Comment:
Similar to how TileLayerConfiguration methods are handled in XMLConfiguration, anything that overrides a BlobStoreConfiguration method should include a javadoc of the form:
```
/**
 * @see BlobStoreConfiguration#getBlobStores()
 */
```